modules/core/shared/src/main/scala-3.0-/eu/timepit/refined/boolean.scala

```diff
@@ -273,6 +273,15 @@ private[refined] trait BooleanInference0 extends BooleanInference1 {
   implicit def conjunctionCommutativity[A, B]: (A And B) ==> (B And A) =
     Inference.alwaysValid("conjunctionCommutativity")
 
+  implicit def substitutionInConjunction[A, B, C](implicit p1: B ==> C): (A And B) ==> (A And C) =
+    p1.adapt("substitutionInConjunction(%s)")
+
+  implicit def disjunctionTautologyElimination[A, B, C](implicit
+      p1: B ==> C,
+      p2: A ==> C
+  ): (A Or B) ==> C =
+    Inference.combine(p1, p2, "disjunctionElimination")
```

Comment:
`Combine` is "and", did you mean to make this use "or"? (as we discussed, probably rqeuires a new method)

Reply:
Yep, but we were wrong :) In order to imply that `A Or B` can be replaced by `C`, both `A` and `C` needs to be equivalents of `C`.


Here is the truth table for OR option:

```
  A B C  │  ((B ↔ C) ∨ (A ↔ C)) → ((A ∨ B) ↔ C)
  ───────┼─────────────────────────────────────
  1 1 1  │      1    1    1    *1     1    1   
  1 1 0  │      0    0    0    *1     1    0   
  1 0 1  │      0    1    1    *1     1    1   
  1 0 0  │      1    1    0    *0     1    0   
  0 1 1  │      1    1    0    *1     1    1   
  0 1 0  │      0    1    1    *0     1    0   
  0 0 1  │      0    0    0    *1     0    0   
  0 0 0  │      1    1    1    *1     0    1  
```